### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by EstebanDalelR**
## Description
Allows logging in to watermelon and storing the credentials for later use
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  
## Acceptance

- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by baristaGeek — COMMENTED** [review]
From what I understand rendering information on the Tool Window Factory will depend on logging in (action which LGTM). If this is the case:

1. I'd like us to re-test after we merge the backend changes
2. The Tool Window Factory should show a text that invites to log in .

File: src/main/kotlin/com/watermelon/context/actions/LoginAction.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            connection.disconnect()
+        }
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        // Open webpage
+        BrowserUtil.browse("$backendUrl/intellij")
+
+        // Open a dialog for user input
+        val email = Messages.showPasswordDialog(
```
Comment (baristaGeek) [inline src/main/kotlin/com/watermelon/context/actions/LoginAction.kt]: Can we please change this from a PasswordDialog to a normal showInputDialog?

File: src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt [hunk 2]
```diff
@@ -39,41 +46,201 @@ class MyToolWindowFactory : ToolWindowFactory {
     override fun shouldBeAvailable(project: Project) = true
 
     class MyToolWindow(toolWindow: ToolWindow) {
+        val backendUrl = "https://app.watermelontools.com"
```
Comment (baristaGeek) [inline src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt:49]: I tested this both with the prod and the localhost backendUrl. In both cases I get this 404 page. 

Uploading Screen Recording 2023-09-14 at 10.50.11 AM.mov…

Comment (baristaGeek) [reply src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt:49]: https://github.com/watermelontools/watermelon-intellij/assets/8325094/92ff52f5-0969-420f-aa38-c5652e5949f6

**Review by baristaGeek — CHANGES_REQUESTED** [review]
We need to 
- Change the user email prompt from a password dialog to an input dialog
- Solve the 404 issue

File: src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt (hunk 2 above)
Comment (EstebanDalelR) [reply src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt:49]: I see it even in prod
https://app.watermelontools.com/intellij

**Comment by EstebanDalelR** [comment]
> We need to
> 
>     * Change the user email prompt from a password dialog to an input dialog
> 
>     * Solve the 404 issue
I changed the input.
The 404 can't be reproduced, not in localhost or prod version of the backend

File: src/main/kotlin/com/watermelon/context/actions/LoginAction.kt (hunk 1 above)
Comment (EstebanDalelR) [reply src/main/kotlin/com/watermelon/context/actions/LoginAction.kt]: Done

File: src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt (hunk 2 above)
Comment (EstebanDalelR) [reply src/main/kotlin/com/watermelon/context/toolWindow/MyToolWindowFactory.kt:49]: Can't reproduce.

**Review by baristaGeek — APPROVED** [review]

**Comment by baristaGeek** [comment]
> > We need to
> > ```
> > * Change the user email prompt from a password dialog to an input dialog
> > 
> > * Solve the 404 issue
> > ```
> 
> I changed the input. The 404 can't be reproduced, not in localhost or prod version of the backend

Ok LGMT then. I'll assume it's something with my local env.

**Pull request merged**